lib/internal/modules/esm/translators.js

```diff
@@ -63,16 +64,27 @@ function errPath(url) {
   return url;
 }
 
-function initializeImportMeta(meta, { url }) {
-  meta.url = url;
-}
-
 let esmLoader;
 async function importModuleDynamically(specifier, { url }) {
   if (!esmLoader) {
-    esmLoader = require('internal/process/esm_loader');
+    esmLoader = require('internal/process/esm_loader').ESMLoader;
+  }
+  return esmLoader.import(specifier, url);
+}
+
+async function importMetaResolve(specifier, parentUrl = this.url) {
```

Comment:
We should definitely document where `this.url` is coming from (`import.meta`). I think I'd also prefer not depending on the receiver at all, since `const { xyz } = import.meta` is probably not going to be a wholly uncommon pattern.

Reply:
Sure, I've fixed this up to use a proper closure. The idea of a function closure per module still abhors me, but I will get over it.